packages/blade/src/components/Checkbox/_decisions/decisions.md

```diff
@@ -0,0 +1,217 @@
+| name (html native)  | `string`   | No       | undefined | The name of the input field in a checkbox, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name)            |
+| value (html native) | `string`   | No       | undefined | The value of the input field in a checkbox, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#value) |
+| isDisabled          | `boolean`  | No       | `false`   | Control whether the checkbox is disabled or not.                                                                                                         |
+| hasError           | `boolean`  | No       | `false`   | Control whether the checkbox is invalid or not.                                                                                                          |
+| isRequired          | `boolean`  | No       | `false`   | Control whether the checkbox is required or not.                                                                                                         |
+
+### Examples:
+
+#### Basic
+
+```tsx
+// Label text
+<Checkbox>is blade awesome?</Checkbox>
+// Helper text
+<Checkbox helpText="Hint: yes">is blade awesome?</Checkbox>
+// uncontrolled
+<Checkbox defaultChecked={true}>is blade awesome?</Checkbox>
+// controlled
+<Checkbox isChecked={true | false} onChange={isChecked => {}}>is blade awesome?</Checkbox>
+```
+
+#### Validations
+
+And `hasError` will pass `aria-invalid` attributes to indicate invalid state.
+
+```tsx
+<Checkbox hasError={true | false}>is blade awesome?</Checkbox>
+```
+
+## `CheckboxGroup` API
+
+| Prop          | Type                        | Required | Default     | Description                                                                                                                        |
+| ------------- | --------------------------- | -------- | ----------- | ---------------------------------------------------------------------------------------------------------------------------------- |
+| children      | `React.ReactNode`             | Yes       | `undefined` | Accepts multiple Checkboxes                                                                                                        |
+| label         | `string`                    | Yes       | `undefined` | The label of the group                                                                                                             |
+| labelPosition | `top, left`                 | No       | `top`       | The position of the rendered label                                                                                                 |
+| helpText    | `string`                    | No       | `undefined` | The helper text to be rendered                                                                                                     |
```

Comment:
I am assuming the only difference between `helpText` and `errorText` at the Checkbox group level is that one appears in grey while the other appears in red, they both have the same positioning. What happens if a user provides values for both? Are they going to stack up on one another. Don't remember seeing that in the designs though.

Reply:
> helpText and users can themselves do helpText = hasError? errortext:helptext?

In here we already have the information if the CheckboxGroup is in error state or not via hasError. and you also need this to change the helper text color right? Users should not handle all these logic IMO. 

So this will look kinda odd: 
<CheckboxGroup hasError helperText={hasError ? "Err" : "Ok" } />

Reply:
Will renaming it to something generic help? like `caption={hasError ? 'boooo' : 'yay'}` sort of API? other alternatives for names `subText`, `text`, `validityMessage`, etc 🤔 ?

 > why do we have errorText at Checkbox group only and not individual checkbox level? is that intentional?

cause I also got confused in this part. I would imagine API on all components from design system to be similar otherwise people will have to remember each and every decision while working no?

Reply:
>  like caption={hasError ? 'boooo' : 'yay'}

yes but we will still need to know the error state. to be able to set proper aria-invalid & colors. 

> cause I also got confused in this part

We've changed this, now single checkbox will also have errorText